resdk/resources/collection.py

```diff
@@ -28,11 +28,14 @@ class BaseCollection(BaseResource):
     WRITABLE_FIELDS = ('data', 'description', 'settings', 'descriptor_schema',
```

Comment:
I guess your tests did not cover all the bordering cases. What would happen now if you get the data, change one of the hydrated data objects and then call `save()` on the collection? Or if a user appends an ID to the collections of hydrated data objects and then calls `save()` on the collection? ...